objectherkenning_openbare_ruimte/on_edge/detection_pipeline/components/model_result.py

```diff
@@ -0,0 +1,314 @@
+import logging
+import os
+import pathlib
+from typing import List, Tuple, Union
+
+import cv2
+import numpy as np
+import numpy.typing as npt
+from ultralytics.engine.results import Results
+
+logger = logging.getLogger("model_result")
+
+
+class ModelResult:
+    def __init__(
+        self, model_result: Results, target_classes, sensitive_classes
+    ) -> None:
+        self.result = model_result.cpu()
+        self.image = self.result.orig_img.copy()
+        self.boxes = self.result.boxes.numpy()
+        self.target_classes = target_classes
+        self.sensitive_classes = sensitive_classes
+
+    def process_detections_and_blur_sensitive_data(
+        self, image_detection_path: str, image_file_name: pathlib.Path
+    ):
+        for summary_str in self._yolo_result_summary():
+            logger.info(summary_str)
+
+        target_idxs = np.where(np.in1d(self.boxes.cls, self.target_classes))[0]
+        if len(target_idxs) == 0:
+            logger.debug("No container detected, not storing the image.")
+            return False
+
+        sensitive_idxs = np.where(np.in1d(self.boxes.cls, self.sensitive_classes))[0]
+        if len(sensitive_idxs) > 0:
+            sensitive_bounding_boxes = self.boxes[sensitive_idxs].xyxy
+            self.blur_inside_boxes(sensitive_bounding_boxes)
+
+        target_bounding_boxes = self.boxes[target_idxs].xyxy
+        self.draw_bounding_boxes(target_bounding_boxes)
+
+        self.save_image(image_detection_path, image_file_name)
+        self.save_annotation_path(target_idxs, image_detection_path, image_file_name)
+
+        return True
+
+    def save_image(self, image_detection_path, image_file_name):
```

Comment:
Why do we log here and not in the `save_annotation_path` method below? Maybe easier to follow / debug if we save both in the same method. Especially since both share two input parameters.